File: write_into_mcu.c
#include <stdio.h>
#include "string.h"

#define RESFILE "result.txt"
#define PREFILE "C:\\Users\\30744\\Desktop\\ComputerVision\\CV.txt"
#define ISPFILE "C:\\CodeFiles\\MCU_C51\\CV.c"
#define target_line 8

int main() {
    FILE *f = fopen(RESFILE, "r");
    char num = 0;
    int line = 0;
    fscanf(f, "%c", &num);
    fclose(f);
    if (num != 0) {
        FILE *f_old = fopen(PREFILE, "r");
        FILE *f_new = fopen(ISPFILE, "w");
        char *msg = "unsigned char timer1=";

        if (f_new != NULL && f_old != NULL) {
            printf("%s", "ok");
            char file_msg[100];
            while (fgets(file_msg, sizeof(file_msg), f_old)) {
                line++;
                if (line == target_line) {
                    fprintf(f_new, "%s%c%c%c", msg, num, ';', '\n');
                }
                else {
                    fprintf(f_new, "%s", file_msg);
                }
                memset(file_msg, '\0', sizeof(file_msg));
            }
        }
        fclose(f_new);
    }

}

File: write_into_mcu.py
# 打开目标文件
res_file = open(r"C:\Users\30744\Desktop\ComputerVision\result.txt", 'r')
pre_file = open(r"C:\Users\30744\Desktop\ComputerVision\CV.txt", 'r')
isp_file = open(r"C:\Users\30744\Desktop\ComputerVision\mcu.c", 'w')
target_line = 8

# 读取结果文件和应该写入mcu.c的文件，并根据读到的result对后者进行修改后写入
result = res_file.read(1)
pretext = pre_file.readlines()
pretext[target_line] = 'unsigned char timer1=' + result + ';' + '\n'

# 写入mcu.c
for i in range(0, len(pretext)):
    isp_file.write(pretext[i])